### ISSUE TITLE
Idea to consider: Setup OWN CI-server with own F-Droid repo

### ISSUE THREAD

**Issue by agilob** [SecUpwN/Android-IMSI-Catcher-Detector#513]
Leave buildozer, setup own Gitlab-ci or Jenkins to build each commit and publish apk for every build in own F-Droid repo. We could provide updates for every minor bug in own repository and major releases in standard way - official FDroid repo.

**Comment by DJaeger** [comment]
Great idea, have thought about too.

**Comment by E3V3A** [comment]
Thank you! We didn't know about that possibility. But this will have to be done after #215.

**Comment by DJaeger** [comment]
How does this depend on #215 ?
As this requires other knowledge than a database rework, others can work on this.
Am I wrong?

**Comment by E3V3A** [comment]
This sound like an extremely low priority item, so if you mean by "others", other developers than what we already have, then that's just fine.

**Comment by DJaeger** [comment]
This is for me more a administration task than a development task.
So this shouldn't steel development capacities.
Also it requires someone to keep the system running, this is not a onetime task.

**Comment by SecUpwN** [comment]
Great. So now that we solved #453, we shaall now leave Buildozer? Guys, I feel we're running in circles! @agilob, what's your problem with Buildozer? If there would be a way of telling users to "push that button and grab the latest Buildozer build", would you guys also agree with that as a solution for this?

**Comment by DJaeger** [comment]
The adventage of a F-Droid Repo would be, that we don't have to take care of these independant buildnumber and clean up the code by reversing #453.
So we can use our usual buildnumber/versionnames/versioncodes.
Also this would allow to be notified for an update and can update with one click.
And all of that with open source!

**Comment by SecUpwN** [comment]
I will now wait for @He3556 to leave his comment here. I am really not sure what to think of reversing #453 and now having to manage our own F-Droid store and set up another service for this...

**Comment by agilob** [comment]
* FDroid can release new update based on:
 version code
version name 
build date

so we can update app in repository on every merged PR
- buildozer requires account (I still don't have one and I don't want it)
- can't update with one click
- different signatures - uninstall app/install app = data is lost

**Comment by SecUpwN** [comment]
>* can't update with one click

The [Buildozer app](https://play.google.com/store/apps/details?id=io.buildozer.app) just does that.

**Comment by agilob** [comment]
One more disadvantage then:
\- Google Play on your phone

**Comment by E3V3A** [comment]
Many good arguments here, but I really think we need to stick with *Buildozer* for until we get out of **alpha**. The work needed to mess with this is really not worth it, as we've already seen before, it creates a nightmare for devs and maintainers. 

> - buildozer requires account (I still don't have one and I don't want it)

Everything requires an account... not a good argument.

> - can't update with one click

Yes, that's exactly what it is for.

> - different signatures = uninstall app/install app = data is lost

We're in alpha with many DB changes, so that is exactly what we need.

> - Google Play on your phone

You don't need GP to download/install GP apps. @SecUpwN  Please show them how to do it...

**Comment by E3V3A** [comment]
[Here](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/labels/PRIORITY) is a list of more important topics...

**Comment by agilob** [comment]
>Everything requires an account... not a good argument.

FDroid doesn't.

**Comment by DJaeger** [comment]
Then please leave this open and add a milestone for stable release.
If someone want to setup a F-Droid Repo he could so so, as this does not need to change anything else.
@E3V3A: yes it is possible to install GP apps without GP installed, but do you know if this app is spying you?
Most pleople using AIMSICD and especially who contribute to it, don't want to use closed source apps as they can not determine what it is doing.
